### ISSUE TITLE
Forced/Required Response Feature 

### ISSUE THREAD

**Issue by TutiGomoka** [Watts-Lab/surveyor#56]
(This is a feature that we may not require at the moment as we run our own surveys. But we may need it once we start running other people's surveys) But:

- Implement a feature where we can choose to either force a respondent to respond to a question or leave it as optional
1. Question on what the system should do if a respondent hasn't answered a 'forced response' question? Just prevent from submitting?

**Comment by markwhiting** [comment]
Required is already implemented for all questions by default as a server level setting. 

What I think would be useful here is something like an optional column in CSVs called `required` that defaults to the server value when blank, but can take values of `true` or `false` to select if a field is required or not.

**Comment by KaranSampath** [comment]
I think we would need to reconcile this with the overall `required` check that we already have in place. i.e. Do we want both overall and individual required variables? If not, how would we indicate survey-wide required characteristics (is it useful to have any disclaimer if no questions are required?). I've modified the code to include my interpretation of this, but let me know if it needs changes.

**Comment by markwhiting** [comment]
Can you summarize your policy?

I’d just assume required is on by default for everything and then some things are optional if there’s a column that expresses that.

**Comment by KaranSampath** [comment]
Yes exactly, so each question is required unless specified (will take on that value in that case (i.e. either true or false)).
